packages/graphql/src/schema/subscriptions/generate-subscription-types.ts

```diff
@@ -87,29 +109,161 @@ export function generateSubscriptionTypes({
+                        const trueSource = source as RelationSubscriptionsEvent;
+                        const r = node.relationFields.find((f) => f.type === trueSource.relationshipName)?.fieldName;
+                        return r;
+                    },
+                },
+                direction: {
+                    type: relationDirectionEnum.NonNull,
+                    resolve: (source: SubscriptionsEvent) => {
+                        const trueSource = source as RelationSubscriptionsEvent;
+                        const isThisTo = trueSource.toTypename === node.name;
+                        return isThisTo ? "IN" : "OUT";
+                    },
+                },
+            });
+
+            relationDisconnectedEvent.addFields({
+                [subscriptionEventPayloadFieldNames.disconnect]: {
+                    type: eventPayload.NonNull,
+                    resolve: (source: SubscriptionsEvent) => {
+                        if ((source as RelationSubscriptionsEvent).fromTypename === node.name) {
+                            return (source as RelationSubscriptionsEvent).properties.from;
+                        }
+                        return (source as RelationSubscriptionsEvent).properties.to;
+                    },
+                },
+                relationshipName: {
+                    type: new GraphQLNonNull(GraphQLString),
+                    resolve: (source: SubscriptionsEvent) => {
+                        const trueSource = source as RelationSubscriptionsEvent;
+                        return node.relationFields.find((f) => f.type === trueSource.relationshipName)?.properties;
+                    },
+                },
+                direction: {
+                    type: relationDirectionEnum.NonNull,
+                    resolve: (source: SubscriptionsEvent) => {
+                        const trueSource = source as RelationSubscriptionsEvent;
+                        const isThisTo = trueSource.toTypename === node.name;
+                        return isThisTo ? "IN" : "OUT";
+                    },
+                },
+            });
+        }
+
+        if (_hasProperties(relationsEventPayload)) {
+            const resolveRelationship = (source: SubscriptionsEvent) => {
+                const trueSource = source as RelationSubscriptionsEvent;
+                const isThisTo = trueSource.toTypename === node.name;
+                const props = isThisTo ? trueSource.properties.from : trueSource.properties.to;
+                const typename = isThisTo ? trueSource.fromTypename : trueSource.toTypename;
+
+                const thisRel = node.relationFields.find(
+                    (f) => f.type === trueSource.relationshipName
+                ) as RelationField;
+
+                return {
+                    [thisRel.fieldName]: {
+                        edge: {
+                            ...trueSource.properties.relationship,
+                        },
```

Comment:
I believe we talked about not having `edge` here because the relationship field name is already technically an "edge"? This can maybe be tweaked later with the rest of the changes discussed!

Reply:
resolving this as it will be part of a different PR (containing changes to API and filtering)